tests/generic/test_filedata.py

```diff
@@ -0,0 +1,120 @@
+import unittest
+import os
+import json
+import pandas as pd
+
+from pyiron_base.generic.filedata import FileData, load_file
+from pyiron_base import FileHDFio
+
+
+class TestLoadFile(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.current_dir = os.path.dirname(os.path.abspath(__file__)).replace("\\", "/")
+        hdf5 = FileHDFio(file_name=cls.current_dir + '/test_data')
+        with hdf5.open("content") as hdf:
+            hdf['key'] = "value"
+        hdf5 = FileHDFio(file_name=cls.current_dir + "/test_data.h5")
+        with hdf5.open("content") as hdf:
+            hdf['key'] = "value"
+        with open(cls.current_dir + '/test_data.txt', 'w') as f:
+            f.write("some text")
+        with open(cls.current_dir + '/test_data2', 'w') as f:
+            f.write("some text")
+        with open(cls.current_dir + '/test_data.csv', 'w') as f:
+            f.write("id,status,chemicalformula")
+            f.write("13,aborted,Al108")
+        with open(cls.current_dir + '/test_data.json', 'w') as f:
+            json.dump({'x': [0, 1]}, f)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.current_dir = os.path.dirname(os.path.abspath(__file__)).replace("\\", "/")
+        for file in ['/test_data', "/test_data.h5",  "/test_data.txt",
+                     "/test_data.csv", "/test_data.json", "/test_data2"]:
+            os.remove(cls.current_dir + file)
+
+    def test_load_file_txt(self):
+        txt = load_file(self.current_dir+'/test_data.txt')
+        with open(self.current_dir+'/test_data.txt') as f:
+            content = f.readlines()
+        self.assertEqual(txt, content)
+
+    def test_load_file_csv(self):
+        csv = load_file(self.current_dir+'/test_data.csv')
+        content = pd.read_csv(self.current_dir+'/test_data.csv')
+        self.assertTrue(content.equals(csv))
+
+    def test_load_file_hdf(self):
+        hdf = load_file(self.current_dir+'/test_data.h5')
+        self.assertIsInstance(hdf, FileHDFio)
+        self.assertEqual(hdf['content/key'], 'value')
+
+    def test_load_file_default(self):
+        """Test default load for text file and h5 file without extension."""
+        filename = self.current_dir + '/test_data'
+        default = load_file(filename)
+        # For some reason on linux handles this different than windows:
+        self.assertTrue(filename == default or default[0] == '‰HDF\n')
```

Comment:
I don't understand what that second test is checking for.

Reply:
Here, I actually want to have a failure of a normal `readlines` and thus `load_file` to return the filename. However, on windoof the read of the hdf file returns some cryptic byte like strings and the first one happens to be this... I really would like the windows `readlines` to throw an error on this...